### DISCUSSION TITLE
validateCsrfTokens - does not accept dynamic parameters

### DISCUSSION THREAD

**Discussion by devexchanger** [laravel/framework#51005] [Q&A]
### Laravel Version

11.3

### PHP Version

8.2

### Database Driver & Version

mysql

### Description

Is there no way to add dynamic paths to validateCsrfTokens?
They are very useful for work.

Now have to write everything manually.
For example, if the path to the administrator is created dynamically, have to manually change here

### Steps To Reproduce

```php
return Application::configure(basePath: dirname(__DIR__))
    ->withProviders()
    ->withMiddleware(function (Middleware $middleware) {

      $middleware->validateCsrfTokens(except: [
            'test/*',
            config('project.path') // does not work
        ]);
  });
```

**Comment by devajmeireles** [comment]
This happens because the `config` function has not yet been loaded/registered when the application is being created.

**Comment by erikverbeek** [comment]
For anyone who might come across this question, like I did. You can add the exception rules via the AppServiceProvider.

```php
<?php

namespace App\Providers;

use Illuminate\Support\ServiceProvider;

class AppServiceProvider extends ServiceProvider
{
    public function boot(): void
    {
        \Illuminate\Foundation\Http\Middleware\VerifyCsrfToken::except([
            'test/*',
            config('project.path') // Should work here
        ]);
    }
}
```

At this point in the lifecycle the config function has been loaded.
Credit goes to this post: https://stackoverflow.com/questions/78279024/laravel-11-disable-csrf-for-a-route
